From 090f090b6351f5f1b9526d214aa37f2ed98b5d9d Mon Sep 17 00:00:00 2001
From: Ben Peled <bpeled@marvell.com>
Date: Wed, 19 May 2021 18:59:29 +0300
Subject: [PATCH 1565/1921] dts: cp110: eip197: add cell index property

This patch adds cell-index property which allows to distinguish
between multiple engines in devices with 2 EIP197 engines.

Binding document was updated accordingly.

Change-Id: I5e7077d1af3b5cec117ec01c33c8136e5207daca
Signed-off-by: Igal Liberman <igall@marvell.com>
Signed-off-by: Ben Peled <bpeled@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52166
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index 96084bf63758..75220c9c2e0b 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -12,6 +12,7 @@
 
 #define CP11X_PCIEx_CONF_BASE(iface)	(CP11X_PCIEx_MEM_BASE(iface) + CP11X_PCIEx_MEM_SIZE(iface))
 #define CP11X_PCIEx_EP_REG_BASE(iface)	(ADDRESSIFY(CP11X_BASE) + 0x600000 + (iface) * 0x4000)
+#define CP11X_EIP197_INDEX		CP11X_NUM
 #define CP11X_PCIEx_MAC_RESET_BIT(n)	(11 + ((n + 2) % 3))
 
 / {
@@ -556,6 +557,7 @@
 			clock-names = "core", "reg";
 			clocks = <&CP11X_LABEL(clk) 1 26>,
 				 <&CP11X_LABEL(clk) 1 17>;
+			cell-index = <CP11X_EIP197_INDEX>;
 			dma-coherent;
 		};
 
-- 
2.31.1

